File: LeetCode/Medium/0238-product-of-array-except-self/0238-product-of-array-except-self.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        final int SIZE = nums.length;
        int[] result = new int[SIZE];
        
        int left = 1;
        for (int i = 0; i < SIZE; i++) {
            result[i] = left;
            left *= nums[i];
        }
        
        int right = 1;
        for (int i = SIZE - 1; i >= 0; i--) {
            result[i] *= right;
            right *= nums[i];
        }
        
        return result;
    }
}

File: LeetCode/Medium/0238-product-of-array-except-self/0238-product-of-array-except-self.kt
class Solution {
    fun productExceptSelf(nums: IntArray): IntArray {
        val answer = IntArray(nums.size)

        var k = 1
        for (i in nums.indices) {
            answer[i] = k
            k *= nums[i]
        }

        k = 1
        for (i in nums.indices.reversed()) {
            answer[i] *= k
            k *= nums[i]
        }

        return answer
    }
}